Date: 2005-06-26
Initial Package Version: 2.9.6
Upstream Status: Not yet submitted
Origin: Marcus Habermehl <bmh1980de@yahoo.de>
Description: Adds a logging feature to makepkg.
diff -Naur pacman-2.9.6.orig/doc/makepkg.8.in pacman-2.9.6/doc/makepkg.8.in
--- pacman-2.9.6.orig/doc/makepkg.8.in	2005-06-26 22:08:32.000000000 +0200
+++ pacman-2.9.6/doc/makepkg.8.in	2005-06-26 22:10:36.000000000 +0200
@@ -390,6 +390,9 @@
 Sets MAKEFLAGS="-j<jobs>" before building the package.  This is useful for overriding
 the MAKEFLAGS setting in /etc/makepkg.conf.
 .TP
+.B "\-k, \-\-logging"
+Logging package build process
+.TP
 .B "\-l <pkgname>"
 Download the package's buildscript before starting the build.  This is useful
 if you do not want to do a full 'repoman upd' nor want to download manually the
diff -Naur pacman-2.9.6.orig/scripts/makepkg pacman-2.9.6/scripts/makepkg
--- pacman-2.9.6.orig/scripts/makepkg	2005-06-26 22:08:32.000000000 +0200
+++ pacman-2.9.6/scripts/makepkg	2005-06-26 22:17:40.000000000 +0200
@@ -189,6 +189,7 @@
 	echo "  -h, --help       This help"
 	echo "  -i, --install    Install package after successful build"
 	echo "  -j <jobs>        Set MAKEFLAGS to \"-j<jobs>\" before building"
+	echo "  -k, --logging    Logging package build process"
 	echo "  -l <pkgname>     Download buildscipts before building"
 	echo "  -m, --nocolor    Disable colorized output messages"
 	echo "  -n, --nostrip    Do not strip binaries/libraries"
@@ -209,6 +210,7 @@
 CLEANUP=0
 CLEANCACHE=0
 INSTALL=0
+LOGGING=0
 DOWNLOAD=""
 GENMD5=0
 DEP_BIN=0
@@ -239,6 +241,7 @@
 		--nocolor)    USE_COLOR="n" ;;
 		--genmd5)     GENMD5=1 ;;
 		--rmdeps)     RMDEPS=1 ;;
+		--logging)    LOGGING=1 ;;
 		--noup2date)     NOUP2DATE=1 ;;
 		--help)
 			usage
@@ -249,7 +252,7 @@
 			exit 1
 			;;
 		-*)
-			while getopts "acCsbdehifgj:l:mnorp:uw:-" opt; do
+			while getopts "acCsbdehifgj:l:kmnorp:uw:-" opt; do
 				case $opt in
 					a) SEARCHDEPS=1 ;;
 					c) CLEANUP=1 ;;
@@ -265,6 +268,7 @@
 						;;
 					i) INSTALL=1 ;;
 					j) export MAKEFLAGS="-j$OPTARG" ;;
+					k) LOGGING=1 ;;
 					l) DOWNLOAD="$OPTARG" ;;
 					m) USE_COLOR="n" ;;
 					n) NOSTRIP=1 ;;
@@ -636,9 +640,13 @@
 # build
 msg "Starting build()..."
 stime=`date +%s`
-build 2>&1
+BUILDLOG=${startdir}/${pkgname}-${pkgver}-${pkgrel}-${CARCH}.log
+build 2>&1 | tee "${BUILDLOG}"
 if [ $? -gt 0 ]; then
 	error "Build Failed.  Aborting..."
+	if [ "${LOGGING}" == "0" ] ; then
+		rm -f "${BUILDLOG}"
+	fi
 	exit 2
 fi
 
@@ -825,6 +833,27 @@
 	pacman -R $makedeplist $deplist
 fi
 
+# check optimalization and write it to ${BUILDSCRIPT}
+msg "Checking for optimalization..."
+if grep -q -- "\($CFLAGS\|$CXXFLAGS\)" "${BUILDLOG}" ; then
+	if ! grep -q "# optimalization" "${BUILDSCRIPT}" ; then
+		echo -e "\n# optimalization OK" >> "${BUILDSCRIPT}"
+	fi
+else
+	warning "This package isn't ${CARCH} optimized!"
+fi
+
+# add vim ft line if missing
+if ! grep -q "vim: ft=sh" "${BUILDSCRIPT}" ; then
+	echo -e "\n# vim: ft=sh" >> "${BUILDSCRIPT}"
+fi
+
+# remove build log
+if [ "${LOGGING}" == "0" ] ; then
+	msg "Removing build log..."
+	rm -f "${BUILDLOG}"
+fi
+
 # update last mod line
 sed -i "s/^\(# Last modified: \).*/\1`date -R`/i" $BUILDSCRIPT
 
